app/jobs/in_person/send_proofing_notification_job.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module InPerson
+  class SendProofingNotificationJob < ApplicationJob
+    # @param [Number] enrollment_id primary key of the enrollment
+    def perform(enrollment_id)
+      return unless IdentityConfig.store.in_person_proofing_enabled &&
+                    IdentityConfig.store.in_person_send_proofing_notifications_enabled
+      enrollment = InPersonEnrollment.find_by(
+        { id: enrollment_id },
+        include: [:notification_phone_configuration, :user],
+      )
+      return unless enrollment
+      # skip
+      if enrollment.skip_notification_sent_at_set?
+        # log event
+        analytics(user: enrollment.user).
+          idv_in_person_usps_proofing_results_notification_job_skipped(
+            enrollment_code: enrollment.enrollment_code,
+            enrollment_id: enrollment.id,
+          )
+        return
+      end
+      analytics(user: enrollment.user).
+        idv_in_person_usps_proofing_results_notification_job_started(
+          enrollment_code: enrollment.enrollment_code,
+          enrollment_id: enrollment.id,
+        )
+
+      # send notification and log result
+      phone = enrollment.notification_phone_configuration.formatted_phone
+      message = notification_message(enrollment: enrollment)
+      response = Telephony.send_notification(
+        to: phone, message: message,
+        country_code: Phonelib.parse(phone).country
+      )
+      handle_telephony_result(enrollment: enrollment, phone: phone, telephony_result: response)
+
+      # if notification sent successful
+      enrollment.update(notification_sent_at: Time.zone.now) if response.success?
+    ensure
+      unless enrollment.present?
+        enrollment = InPersonEnrollment.find_by(
+          { id: enrollment_id },
+          include: [:notification_phone_configuration,
+                    :user],
+        )
+      end
+      analytics(user: enrollment&.user).
+        idv_in_person_usps_proofing_results_notification_job_completed(
+          enrollment_code: enrollment.enrollment_code, enrollment_id: enrollment&.id,
+        )
```

Comment:
If the enrollment isn't present by this point, then we should log the completion with a note that an error occurred. We should also avoid trying to re-fetch the enrollment.

Reply:
@NavaTim, this is not a refetch, since it's in final ensure block, the enrollment may not be fetched if the flags are disabled(first two lines).

Reply:
If that's what you intend, then I recommend moving the `find_by` to the section where the flags are disabled. Also note that the accessor for `enrollment_code` here isn't using the nil-safe operator.

Reply:
@NavaTim , refactor it a little bit, logging error and use safe operator.